asf_tools/composite.py

```diff
@@ -0,0 +1,306 @@
+        start = start + step
+
+
+def get_full_extent(corners):
+    """"Calculate the union of corners"""
+    min_ulx = 50000000
+    max_lrx = 0
+    max_uly = 0
+    min_lry = 50000000
+
+    for fi, ulx, lrx, lry, uly in corners:
+        logging.debug(f"{ulx,uly} {lrx,lry}")
+        min_ulx = min(ulx, min_ulx)
+        max_uly = max(uly, max_uly)
+        max_lrx = max(lrx, max_lrx)
+        min_lry = min(lry, min_lry)
+
+    logging.debug(f"Return is upper left: {min_ulx,max_uly}; lower right: {max_lrx,min_lry}")
+    return min_ulx, max_lrx, max_uly, min_lry
+
+
+def get_max_pixel_size(files):
+    """Find maximum pixel size of given files"""
+    pix_size = -999
+    for fi in files:
+        (x1, y1, t1, p1) = saa.read_gdal_file_geo(saa.open_gdal_file(fi))
+        tmp = t1[1]
+        pix_size = max(pix_size, tmp)
+
+    if pix_size == -999:
+        raise Exception("No valid pixel sizes found")
+    return pix_size
+
+
+def get_hemisphere(fi):
+    """Return hemisphere of UTM zone - North or South"""
+    hemi = None
+    dst = gdal.Open(fi)
+    p1 = dst.GetProjection()
+    ptr = p1.find("UTM zone ")
+    if ptr != -1:
+        (zone, hemi) = [t(s) for t, s in zip((int, str), re.search('(\d+)(.)', p1[ptr:]).groups())]
+    return hemi
+
+
+def get_zone_from_proj(fi):
+    """Return the UTM zone of given file"""
+    zone = None
+    dst = gdal.Open(fi)
+    p1 = dst.GetProjection()
+    ptr = p1.find("UTM zone ")
+    if ptr != -1:
+        (zone, hemi) = [t(s) for t, s in zip((int, str), re.search("(\d+)(.)", p1[ptr:]).groups())]
+    return zone
+
+
+def parse_zones(files):
+    """Return the zone numbers of all files given"""
+    zones = []
+    for fi in files:
+        zone = get_zone_from_proj(fi)
+        if zone:
+            zones.append(zone)
+    return np.asarray(zones, dtype=np.int8)
+
+
+def reproject_to_median_utm(files, pol, resolution=None):
+    """Reproject a bunch of UTM geotiffs to the median UTM zone.
+       Use either the given resolution or the largest resolution in the stack"""
+
+    if len(files) < 2:
+        return None
+
+    # Set the pixel size
+    if resolution:
+        pix_size = resolution
+        logging.info(f"Changing pixel size to {pix_size}")
+    else:
+        pix_size = get_max_pixel_size(files)
+        logging.info(f"Using maximum pixel size {pix_size}")
+
+    # Get the median UTM zone and hemisphere
+    home_zone = np.median(parse_zones(files))
+    logging.info(f"Home zone is {home_zone}")
+    hemi = get_hemisphere(files[0])
+    logging.info(f"Hemisphere is {hemi}")
+
+    # Reproject files as needed
+    logging.info("Checking projections")
+    new_files = []
+    for fi in files:
+        fi_name = os.path.split(fi)[1]
+        my_zone = get_zone_from_proj(fi)
+        name = fi_name.replace(".tif", "_reproj.tif")
+        afi = fi.replace(f"_{pol}.tif", "_area.tif")
+        aname = fi_name.replace(f"_{pol}.tif", "_area_reproj.tif")
+        if not os.path.isfile(name):
```

Comment:
Need to check area file here too; running VV and then VH fails when trying to make the already existing area link